solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,523 @@
+                grossNetworkTokenConversionAmount = grossNetworkTokenConversionAmount.add(networkTokenConversionAmount);
+            }
+        }
+
+        // calculate the burn incentive fee and reduce it from the total amount to convert
+        (uint256 netNetworkTokenConversionAmount, uint256 incentiveFeeAmount) =
+            applyIncentiveFee(grossNetworkTokenConversionAmount);
+
+        // convert all network token amounts to the governance token
+        totalGovTokenAmountToBurn = totalGovTokenAmountToBurn.add(
+            network.rateByPath(govPath, netNetworkTokenConversionAmount)
+        );
+        require(totalGovTokenAmountToBurn > 0, "ERR_ZERO_TARGET_AMOUNT");
+
+        return (amounts, networkTokenConversionAmounts, totalGovTokenAmountToBurn, incentiveFeeAmount);
+    }
+
+    /**
+     * @dev transfers the ownership of the network fee wallet
+     *
+     * @param newOwner the new owner of the network fee wallet
+     */
+    function transferNetworkFeeWalletOwnership(address newOwner) external ownerOnly {
+        networkFeeWallet().transferOwnership(newOwner);
+    }
+
+    /**
+     * @dev accepts the ownership of he network fee wallet
+     */
+    function acceptNetworkFeeOwnership() external ownerOnly {
+        networkFeeWallet().acceptOwnership();
+    }
+
+    /**
+     * @dev returns the vortex conversion strategy for the specified tokens
+     *
+     * @param tokens the tokens to convert
+     *
+     * @return the conversion paths for each of tokens to convert
+     * @return the conversion amounts for each of tokens to convert
+     * @return the conversion path for the last network to governance tokens conversion
+     * @return the count of the elements in the strategy
+     */
+    function vortexStrategy(IERC20[] calldata tokens, ITokenHolder feeWallet)
+        private
+        view
+        returns (
+            address[][] memory,
+            uint256[] memory,
+            address[] memory,
+            uint256
+        )
+    {
+        IConverterRegistry registry = converterRegistry();
+
+        // create conversion paths and collect available token amounts
+        address[][] memory paths = new address[][](tokens.length);
+        uint256[] memory amounts = new uint256[](tokens.length);
+
+        uint256 count = 0;
+        for (uint256 i = 0; i < tokens.length; ++i) {
+            IERC20 token = tokens[i];
+
+            // make sure to retrieve the balance of either an ERC20 or an ETH reserve
+            uint256 amount;
+            if (token == NATIVE_TOKEN_ADDRESS) {
+                amount = address(feeWallet).balance;
+            } else {
+                amount = token.balanceOf(address(feeWallet));
+            }
+
+            // avoid empty conversions
+            if (amount == 0) {
+                continue;
+            }
+
+            // increment the count of the elements in the strategy
+            count++;
+
+            amounts[i] = amount;
```

Comment:
Shouldn't this be `amounts[count]` though (and before you increment `count`)?

Reply:
Yes, you're right